packages/citizen-sdk/src/sdks/viem-claim-sdk.ts

```diff
@@ -0,0 +1,368 @@
+      hash,
+      retryDelay: 5000,
+    })
+  }
+
+  /**
+   * Checks if the connected user is eligible to claim UBI for the current period.
+   * Returns the amount they can claim (0 if not eligible or already claimed).
+   * Does not check for whitelisting status.
+   * @param pClient - Optional public client to check entitlement on alternative chain.
+   * @returns The claimable amount in the smallest unit (e.g., wei).
+   * @throws If the entitlement check fails.
+   */
+  async checkEntitlement(pClient?: PublicClient): Promise<bigint> {
+    return this.readContract<bigint>(
+      {
+        address: !pClient ? this.ubiSchemeAddress : this.ubiSchemeAltAddress,
+        abi: ubiSchemeV2ABI,
+        functionName: "checkEntitlement",
+        args: [this.account],
+      },
+      pClient,
+    )
+  }
+
+  /**
+   * Attempts to claim UBI for the connected user.
+   * 1. Checks if the user is whitelisted using IdentitySDK.
+   * 2. If not whitelisted, redirects to face verification and throws an error.
+   * 3. If whitelisted, checks if the user can claim UBI by ensuring sufficient balance.
+   * 4. If the user cannot claim due to low balance, triggers a faucet request and waits.
+   * 5. If whitelisted and can claim, proceeds to call the claim function on the UBIScheme contract.
+   * @returns The transaction receipt if the claim is successful.
+   * @throws If the user is not whitelisted, balance check fails, or claim transaction fails.
+   */
+  async claim(): Promise<TransactionReceipt | any> {
+    const userAddress = this.account
+
+    // 1. Check whitelisting status
+    const { isWhitelisted } =
+      await this.identitySDK.getWhitelistedRoot(userAddress)
+    if (!isWhitelisted) {
+      await this.fvRedirect()
+      throw new Error("User requires identity verification.")
+    }
+
+    // 2. Ensure the user has sufficient balance to claim
+    const canClaim = await this.checkBalanceWithRetry()
+    if (!canClaim) {
+      throw new Error("Failed to meet balance threshold after faucet request.")
+    }
+
+    // 3. Execute the claim transaction
+    try {
+      return await this.submitAndWait({
+        address: this.ubiSchemeAddress,
+        abi: ubiSchemeV2ABI,
+        functionName: "claim",
+        chain: this.walletClient.chain,
+      })
+    } catch (error: any) {
+      if (error instanceof ContractFunctionExecutionError) {
+        throw new Error(`Claim failed: ${error.shortMessage}`)
+      }
+      throw new Error(`Claim failed: ${error.message}`)
+    }
+  }
+
+  /**
+   * Redirects the user through the face-verification flow.
+   * @throws If face verification redirect fails.
+   */
+  private async fvRedirect(): Promise<void> {
+    const fvLink = await this.identitySDK.generateFVLink(false, this.rdu, 42220)
+    if (typeof window !== "undefined") {
+      window.location.href = fvLink
+    } else {
+      throw new Error(
+        "Face verification redirect is only supported in browser environments.",
```

Comment:
are you explaining in identitysdk the steps how it can be used in native envs (based on webviews)?

Reply:
Will do after merge, I believe this to be a sidecase and not the most used environment